### DISCUSSION TITLE
How to confirm i'm sniffing properly

### DISCUSSION THREAD

**Discussion by maxdd** [rfquack/RFQuack#52] [General]
Hello,
I started to look at this project because i wanted to re a communication protocol running on a [PL51WT020](https://pmicro.com.cn/uploads/files/20231228/65fd36d6ad876dde35df69d48736decb.pdf).
The datasheet reports a 2.4Ghz radio with up to 1Mbit rate so i felt like an esp32 and a nrf24l01+ would be a good starting point.

I took inspiration from the youtube video sniffing 2.4ghz but apparently i can't even sniff my M580 mouse (although it was connected via bluetooth and not via the dongle so it might be that)

I'm seeking support to understand why i can't see much.
I simply did

```
$ > import time
$ > q.radioA.set_modem_config(bitRate=1000, isPromiscuous=True)
$ > q.radioA.rx()
            for f in range(2400, 2483):
                 q.radioA.set_modem_config(carrierFreq=f)
                 time.sleep(0.15)
            q.radioA.idle()
```

My wiring is

```
GND    GND
CE     G27
SCK    G18
MISO   G19
3.3V   3.3V
CSN    G13
MOSI   G23
IRQ    G33
```

platformio.ini has esp32devkit enabled and build.env has the example esp32 nrf24 configuration.

I can only see one message every scan (that feels strange) apparently on different frequencies irrespective whether i spam the remote controller buttons or move my mouse.
Does that mean my HW is at least properly sniffing?

![image](https://github.com/user-attachments/assets/0f7177c6-d463-4ffc-9cf0-200a827d0219)
![image](https://github.com/user-attachments/assets/0fc7f12c-8fcc-4587-a066-d8d8da24dbce)

The messages seems to appear always at 2400 or 2483
I'm under the impression that those are garbage
Regards,

**Comment by Crsarmv7l** [comment]
See here https://github.com/rfquack/RFQuack/issues/44#issuecomment-2096026675

I have upstreamed quite a few things to radiolib, but rfquack seems fairly abandoned and it's not in line with upstream. Likewise I have moved on and I'm developing my own hw/fw (which is why I'm upstreaming).

**Reply by maxdd** [reply]
Would you mine share your tools? I'm on nrf24 though and i was looking for something quite plug and play to focus on the RE part

**Comment by phretor** [comment]
If you folks have changes, I'm happy to merge them.

**Reply by maxdd** [reply]
Are the screenshots I've pasted correct from your pov? What feels strange is that Carrier is 0.0, RSSI is 0, what is that RxRadio set to 0. Also for CS pin shall we use SPI slave select or a dedicated pin? I mean if MISO and MOSI are generally not specified why would CS be?

**Reply by maxdd** [reply]
The registers are set correctly though

![image](https://github.com/user-attachments/assets/38585924-03b1-4f16-a07a-7841e5b162ae)

**Reply by maxdd** [reply]
With RF24 lib i managed to confirm the HW is working
```
Transmission successful! Time to transmit = 535 us. Sent: 0.00
Transmission successful! Time to transmit = 535 us. Sent: 0.01
...
Transmission successful! Time to transmit = 535 us. Sent: 0.26

Received 4 bytes on pipe 1: 0.00
Received 4 bytes on pipe 1: 0.01
...
Received 4 bytes on pipe 1: 0.26
```
I'll try with RFQuack now to send and receive.
Are there examples?

**Reply by maxdd** [reply]
Radiolib blocking RX and TX examples are also working on my setup.
I will try to focus on promiscuous mode now because i doubt my devices (Logitech Unifi) + Custom Remote Controller (PL51WT020) are not adequate to be sniffed

**Reply by maxdd** [reply]
@phretor Hello Federico, would you mind sharing the code to sniff the RadioLib example? I'm struggling a bit to understand what to set. The example is [this](https://github.com/jgromes/RadioLib/tree/master/examples/nRF24/nRF24_Transmit_Blocking) and it is sending "Hello World!" with the following param:

```
freq – Carrier frequency in MHz. Defaults to 2400 MHz.
dr – Data rate to be used in kbps. Defaults to 1000 kbps.
byte addr[] = {0x01, 0x23, 0x45, 0x67, 0x89}; //width 5
```

If i swipe from 2.4 to 2.483 i see nothing

**Reply by phretor** [reply]
Is this related to RFQuack?

**Reply by maxdd** [reply]
Yes with RFQuack I can't pick up any packets, not even those packets that are sent using pure radiolib example which are definitely working since I can transmit and receive just fine when I flash 2 Esp32 with 2 nrf24. My question is related to the lack of an example in RFQuack that shows how to receive or send a 2.4ghz packet

**Reply by phretor** [reply]
Have you watched the mousejacking attack demo with RFQuack on YouTube?

**Reply by maxdd** [reply]
As I've mentioned in initial post I took inspiration from the youtube video sniffing 2.4ghz where you move the mouse and show packets coming in. 
The same approach doesn't work for me (maybe because your preamble is 0xAA00 and I might need 0x5500, I haven't tried) but I wanted to narrow a bit the scope and check whether my setup can send or receive single packets, unfortunately the one I'm sending with RadioLib with the default parameters above is not captured by RFQuack

**Comment by Crsarmv7l** [comment]
A summary of my changes are in the linked post and already upstreamed. The
major takeaway is that PQT needs to be set to 0 with promiscuous.

Aside from that I have ported a few rfcat features, and working on a few
other s for upstream.

**Reply by maxdd** [reply]
Are they relevant for nrf24?

**Reply by Crsarmv7l** [reply]
That's on you to figure out. If it isn't set with a cc1101 you get nothing.

**Reply by maxdd** [reply]
I would say no since we are talking about 2.4ghz nrf24l01 device. Thank you anyway
